src/op_handlers/mul.rs

```diff
@@ -0,0 +1,35 @@
+use u256::{U256, U512};
+
+use crate::address_operands::{address_operands_read, address_operands_store};
+use crate::{opcode::Opcode, state::VMState};
+
+pub fn _mul(vm: &mut VMState, opcode: &Opcode) {
+    let (src0, src1) = address_operands_read(vm, opcode);
+    let src0 = U512::from(src0);
+    let src1 = U512::from(src1);
+    let res = src0 * src1;
```

Comment:
I think it would be wiser to use overflowing multiplication here, even if these are U512 integers, if I'm correct, the overflowing multiplication will return a tuple (result, overflow), where overflow will be true if it has ocurred, I'd also use this overflow boolean value to test for overflow.

Reply:
The problem with using `U256::overflow_mul()` is that in case of actual overflow we don't get the high bits of the resulting `U512` value. 

Thereby (correct me if wrong) we would even then have to cast to `U512` and then multiply `src0` and `src1` together so we can get the high bits.

Reply:
Sorry, I meant U512::overflow_mul. Does your comment still apply?

Reply:
Considering we are operating with two values that at most are `U256::MAX`, it is impossible that the resulting multiplication between them overflow `U512::MAX`.

```rs
    let u256_max = U256::MAX;
    println!("u256_max: {}", u256_max);
    let u512_max = U512::MAX;
    println!("u512_max: {}", u512_max);
    let overflow = U512::max_value() - (U512::from(u256_max) * U512::from(u256_max));
    println!("overflow: {}", overflow);
 ```
 
 outputs:
 ```bash
 u256_max: 115792089237316195423570985008687907853269984665640564039457584007913129639935
u512_max: 13407807929942597099574024998205846127479365820592393377723561443721764030073546976801874298166903427690031858186486050853753882811946569946433649006084095
overflow: 231584178474632390847141970017375815706539969331281128078915168015826259279870
```

Reply:
My bad, you're right, I was strictly thinking about U512 values.